template/CHART_NAME/templates/statefulset.yaml

```diff
@@ -0,0 +1,145 @@
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
+kind: StatefulSet
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: %%COMPONENT_NAME%%
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels: {{ include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: %%COMPONENT_NAME%%
+  serviceName: {{ template "common.names.fullname" . }}
+  {{- if .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy }}
+  updateStrategy: {{- toYaml .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy | nindent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+        app.kubernetes.io/component: %%COMPONENT_NAME%%
+        {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "%%TEMPLATE_NAME%%.serviceAccountName" . }}
+      {{- include "%%TEMPLATE_NAME%%.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.affinity }}
+      affinity: {{- include "common.tplvalues.render" ( dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.affinity "context" $) | nindent 8 }}
```

Comment:
Update this as well to match `deployment.yml` changes (`nodeAffinity`, `podAntiAffinity`, ... )